desc/objectives/linear_objectives.py

```diff
@@ -166,106 +166,1058 @@ def build(self, eq, use_jit=True, verbose=1):
+        Fourier-Zernike R coefficient target sum. If None,
+         uses current sum of Equilibrium's R coefficients.
+         len(target)=1
     weight : float, ndarray, optional
         Weighting to apply to the Objective, relative to other Objectives.
         len(weight) must be equal to Objective.dim_f
-    normalize : bool
-        Whether to compute the error in physical units or non-dimensionalize.
-    normalize_target : bool
-        Whether target should be normalized before comparing to computed values.
-        if `normalize` is `True` and the target is in physical units, this should also
-        be set to True.
-    fixed_boundary : bool, optional
-        True to enforce the boundary condition on flux surfaces,
-        or False to fix the boundary surface coefficients (default).
+    sum_weight : float, ndarray, optional
+        Weights on the coefficients in the sum, should be same length as modes.
+         Defaults to 1 i.e. target = 1*R_111 + 1*R_222...
     modes : ndarray, optional
-        Basis modes numbers [l,m,n] of boundary modes to fix.
-        len(target) = len(weight) = len(modes).
-        If True/False uses all/none of the profile modes.
+        Basis modes numbers [l,m,n] of Fourier-Zernike modes to fix sum of.
+         len(weight) = len(modes).
+        If True uses all of the Equilibrium's modes.
+        Must be either True or specified as an array
     surface_label : float
         Surface to enforce boundary conditions on. Defaults to Equilibrium.surface.rho
     name : str
         Name of the objective function.
 
-
-    Notes
-    -----
-    If specifying particular modes to fix, the rows of the resulting constraint `A`
-    matrix and `target` vector will be re-sorted according to the ordering of
-    `basis.modes` which may be different from the order that was passed in.
     """
 
     _scalar = False
     _linear = True
     _fixed = False
-    _units = "(m)"
-    _print_value_fmt = "Z boundary error: {:10.3e} "
 
     def __init__(
         self,
         eq=None,
         target=None,
-        bounds=None,
         weight=1,
+        sum_weights=None,
+        modes=True,
+        name="Fix Sum Modes R",
         normalize=True,
         normalize_target=True,
-        fixed_boundary=False,
-        modes=True,
-        surface_label=None,
-        name="lcfs Z",
     ):
 
-        self._fixed_boundary = fixed_boundary
         self._modes = modes
-        self._surface_label = surface_label
-        self._args = ["Z_lmn"] if self._fixed_boundary else ["Zb_lmn"]
+        if modes is None or modes is False:
+            raise ValueError(
+                f"modes kwarg must be specified or True with FixSumModesR! got {modes}"
+            )
+        self._sum_weights = sum_weights
+        if target is not None:
+            if target.size > 1:
+                raise ValueError(
+                    "FixSumModesR only accepts 1 target value, please use multiple"
```

Comment:
Yes, I just was trying to get it cleaned up and this was the easiest way to address its current shortcomings

Reply:
OK. It's fine for now since in most cases the user won't be using these objectives directly. I could also see making `SumModes` into a base class for `FixBoundary` and `FixAxis` at some point.

Reply:
Yea that would be eventually useful, but for now since it is behind the NAE utils API I think it is ok, and the FixAxisR/Z are direct replacements for FixBoundaryR/Z so their use is obvious